docs/scripts/api/buildInterfacesDocumentation.ts

```diff
@@ -173,123 +177,55 @@ const parseInterfaceSymbol = async (
-    );
-
-    if (hasDefaultValue) {
-      text += `| ${formattedName} | ${formattedType} | ${formattedDefaultValue} | ${formattedDescription} |\n`;
-    } else {
-      text += `| ${formattedName} | ${formattedType} | ${formattedDescription} |\n`;
-    }
-  });
-
-  return text;
+function getPlanLevel(property: ParsedProperty) {
+  if (property.projects.includes('x-data-grid')) {
+    return '';
+  }
+  if (property.projects.includes('x-data-grid-pro')) {
+    return 'pro';
+  }
+  if (property.projects.includes('x-data-grid-premium')) {
+    return 'premium';
+  }
+  throw new Error(`No valid plan found for ${property.name} property`);
 }
 
-async function generateImportStatement(objects: ParsedObject[], projects: XTypeScriptProjects) {
-  let imports = '```js\n';
+function getDefaultValue(property: ParsedProperty) {
+  const defaultValue = property.tags.default?.text?.[0].text;
+  if (defaultValue === undefined) {
+    return defaultValue;
+  }
+  return escapeCell(defaultValue);
+}
 
+function generateImportStatement(object: ParsedObject, projects: XTypeScriptProjects) {
   const projectImports = Array.from(projects.values())
     .map((project) => {
-      const objectsInProject = objects.filter((object) => {
-        return !!project.exports[object.name];
-      });
-
-      if (objectsInProject.length === 0) {
+      if (!project.exports[object.name]) {
         return null;
       }
 
-      return `import {${objectsInProject.map((object) => object.name)}} from '@mui/${
-        project.name
-      }'`;
+      return `import { ${object.name} } from '@mui/${project.name}'`;
     })
     .filter((el): el is string => !!el)
     // Display the imports from the pro packages above imports from the community packages
     .sort((a, b) => b.length - a.length);
-
-  imports += await prettier.format(projectImports.join('\n// or\n'), {
-    singleQuote: true,
-    semi: false,
-    trailingComma: 'none',
-    parser: 'typescript',
-  });
-  imports += '\n```';
-
-  return imports;
+  return projectImports;
 }
 
-async function generateMarkdown(
-  object: ParsedObject,
-  projects: XTypeScriptProjects,
-  documentedInterfaces: DocumentedInterfaces,
-) {
-  const demos = object.tags.demos;
-  const description = linkify(object.description, documentedInterfaces, 'html');
-  const imports = await generateImportStatement([object], projects);
-
-  let text = `# ${object.name} Interface\n`;
-  text += `<p class="description">${description}</p>\n\n`;
-
-  if (demos && demos.text && demos.text.length > 0) {
-    text += '## Demos\n\n';
-    text += ':::info\n';
-    text += 'For examples and details on the usage, check the following pages:\n\n';
-    demos.text.forEach((demoLink) => {
-      text += demoLink.text;
-    });
-    text += '\n\n:::\n\n';
+function extractDemos(tagInfo: ts.JSDocTagInfo): { demos?: string } {
+  if (!tagInfo || !tagInfo.text) {
+    return {};
   }
+  const demos = tagInfo.text
```

Comment:
Because it was markdown rendering, the interface description used markdown format to add links to demo

Here I parse them to extract the url and the name to render